mitosheet/mitosheet/api/get_imported_files_and_dataframes.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+from copy import copy
+import json
+import os
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+from mitosheet.types import StepsManagerType
+from mitosheet.step_performers.import_steps.simple_import import SimpleImportStepPerformer
+from mitosheet.step_performers.import_steps.excel_import import ExcelImportStepPerformer
+from mitosheet.step_performers.import_steps.dataframe_import import DataframeImportStepPerformer
+from mitosheet.step_performers.import_steps import is_import_step_type
+
+def get_sublist_at_index_from_optional_list(l: Optional[List[Any]], index: int) -> Optional[List[Any]]:
+    if l is None:
+        return None
+        
+    return [l[index]]
+
+
+def get_import_data_with_single_import_list(step_type: str, params: Dict[str, Any]) -> List[Dict[str, Any]]:
+    """
+    Given a step_type that is an import, and the parameters to that import step,
+    this function will create a list of DataframeCreationData that represent
+    parameters to steps that only create a single dataframe.
+
+    This is useful because the frontend allows users to update dataframe imports one 
+    be one - and so having each step only do a single import is convenient for working
+    with the data on the frontend.
+    """
+    if step_type == SimpleImportStepPerformer.step_type():
+        for index, file_name in enumerate(params['file_names']):
+            return [{
+                'step_type': step_type,
+                'params': {
+                    'file_names': [file_name],
+                    'delimeters': get_sublist_at_index_from_optional_list(params.get('delimeters', None), index),
+                    'encodings': get_sublist_at_index_from_optional_list(params.get('encodings', None), index),
+                    'error_bad_lines': get_sublist_at_index_from_optional_list(params.get('error_bad_lines', None), index)
+                }
+            }]
+
+    if step_type == ExcelImportStepPerformer.step_type():
+        for index, sheet_name in enumerate(params['sheet_names']):
+            return [{
+                'step_type': step_type,
+                'params': {
+                    'file_name': params['file_name'],
+                    'sheet_names': [sheet_name],
+                    'has_headers': params['has_headers'],
+                    'skiprows': params['skiprows']
+                }
+            }]
+
+    if step_type == DataframeImportStepPerformer.step_type():
+        for index, df_name in enumerate(params['df_names']):
+            return [{
+                'step_type': step_type,
+                'params': {
+                    'df_names': [df_name],
+                }
+            }]
+
+    return []
+
+
+def get_imported_files_and_dataframes(params: Dict[str, Any], steps_manager: StepsManagerType) -> str:
```

Comment:
Maybe we can call this `get_imported_files_and_dataframes_from_steps_manager` so that its contrasted with `get_imported_files_and_dataframes_from_analysis_name`. I was confused about them in the beginning.